### PR TITLE
chore: build with clang 11

### PR DESCRIPTION
**Pull request by rainingmaster**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,25 +1,16 @@
...
-COPTS = --config=gcc \
-		--copt "-Wno-stringop-overflow" \
-		--copt "-Wno-vla-parameter" \
-		--copt "-Wno-parentheses" \
-		--copt "-Wno-unused-parameter" \
-		--copt "-Wno-range-loop-construct" \
-		--copt "-Wno-uninitialized" \
-		--copt "-Wno-strict-aliasing" \
-		--copt "-fno-strict-aliasing"
+BUILD_ARGS ?=
```
Comment (doujiang24) [inline Makefile:4]: seems we could remove this? since there already have `BUILD_OPTS`, they could do the same thing.

**Review by doujiang24 — COMMENTED** [review]
@rainingmaster Nice job, please also update the `Requirements` in Readme. otherwise, LGTM.

**Comment by doujiang24** [comment]
https://github.com/mosn/envoy-go-extension/issues/19

File: Makefile (hunk 1 above)
Comment (rainingmaster) [reply Makefile]: fix, sry for ignored it

File: README.md [hunk 2]
```diff
@@ -37,7 +37,9 @@ coming soon.
 ## Requirements
 
 1. Go >= 1.18
-2. Gcc >= 11 (clang not supported yet)
+2. Gcc >= 11
+3. Clang >= 11.1.0
+4. Bazel >= 5.1.1
 
 Note: only support gcc yet. [clang support in TODO](https://github.com/mosn/envoy-go-extension/issues/19).
```
Comment (doujiang24) [inline README.md:44]: this line could be removed now.

File: README.md [hunk 3]
```diff
@@ -37,7 +37,9 @@ coming soon.
 ## Requirements
 
 1. Go >= 1.18
-2. Gcc >= 11 (clang not supported yet)
+2. Gcc >= 11
+3. Clang >= 11.1.0
```
Comment (doujiang24) [inline README.md:41]:
```suggestion
2. Gcc >= 11 or Clang >= 11.1.0
```
maybe, combine them should be better?

Comment (rainingmaster) [reply README.md]: done

File: README.md (hunk 2 above)
Comment (rainingmaster) [reply README.md]: done

**Pull request merged**